tests/app/dao/test_service_letter_contact_dao.py

```diff
@@ -200,27 +200,27 @@ def test_archive_letter_contact_does_dissociates_template_defaults_before_archiv
     assert template_default.archived is True
 
 
-@pytest.mark.xfail(reason='Mislabelled for route removal, fails when unskipped.')
+@pytest.mark.skip(reason='Mislabelled for route removal, fails when unskipped.')
 def test_dao_get_letter_contact_by_id(sample_service):
     letter_contact = create_letter_contact(service=sample_service, contact_block='Aberdeen, AB12 23X')
     result = dao_get_letter_contact_by_id(service_id=sample_service.id, letter_contact_id=letter_contact.id)
     assert result == letter_contact
 
 
-@pytest.mark.xfail(reason='Mislabelled for route removal, fails when unskipped.')
+@pytest.mark.skip(reason='Mislabelled for route removal, fails when unskipped.')
 def test_dao_get_letter_contact_by_id_raises_sqlalchemy_error_when_letter_contact_does_not_exist(sample_service):
     with pytest.raises(SQLAlchemyError):
         dao_get_letter_contact_by_id(service_id=sample_service.id, letter_contact_id=uuid.uuid4())
 
 
-@pytest.mark.xfail(reason='Mislabelled for route removal, fails when unskipped.')
+@pytest.mark.skip(reason='Mislabelled for route removal, fails when unskipped.')
 def test_dao_get_letter_contact_by_id_raises_sqlalchemy_error_when_letter_contact_is_archived(sample_service):
     archived_contact = create_letter_contact(service=sample_service, contact_block='Aberdeen, AB12 23X', archived=True)
     with pytest.raises(SQLAlchemyError):
         dao_get_letter_contact_by_id(service_id=sample_service.id, letter_contact_id=archived_contact.id)
 
 
-@pytest.mark.xfail(reason='Mislabelled for route removal, fails when unskipped.')
+@pytest.mark.skip(reason='Mislabelled for route removal, fails when unskipped.')
```

Comment:
I see you updated this from `xfail` to `skip` - I thought for the #1666 we agreed upon `xfail` for the mislabeled tests. I see you updated 12 - I believe I marked around 40 ish total. Did you want me to update the rest to `skip` in a different PR?

Reply:
If it helps, marking a test with `xfail` still runs the test (which sounds like the problem). If you do want to mark the tests `xfail` but not have them run at all, then add: `run=False` so
`@pytest.mark.xfail(run=False, reason='Mislabelled for route removal, fails when run.')`

Reply:
I changed to skip so they were not ran. While I was troubleshooting the deadlock issue I was grabbing at anything that had the potential to cause deadlocks. Tests that were known to fail seemed like low-hanging fruit. 

Were there any that we want to fail? Why was xfail chosen over skip? The particular ones you linked were letter-related.

Reply:
I don't know why `xfail` was chosen over `skip`. Reading the documentation - `skip` makes more sense.